src/Microsoft.FeatureManagement.AspNetCore/FeatureGateEndpointFilter.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.FeatureManagement.AspNetCore
+{
+
+    /// <summary>
+    /// An endpoint filter that requires a feature flag to be enabled.
+    /// </summary>
+    internal class FeatureGateEndpointFilter : IEndpointFilter
+    {
+        public string FeatureName { get; }
```

Comment:
In the attribute implementation this is called `Features`, to keep things consistent I would rename this.